packages/mds-attachment-service/service/helpers/index.ts

```diff
@@ -0,0 +1,143 @@
+/*
+    Copyright 2019 City of Los Angeles.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+import logger from '@mds-core/mds-logger'
+import aws from 'aws-sdk'
+import path from 'path'
+import sharp from 'sharp'
+import { uuid, UnsupportedTypeError, ValidationError } from '@mds-core/mds-utils'
+import { Attachment, AttachmentSummary } from '@mds-core/mds-types'
+import { SerializedBuffers } from '@mds-core/mds-service-helpers'
+
+/* eslint-disable-next-line */
+const multer = require('multer')
+const { env } = process
+const supportedMimetypes = ['image/png', 'image/jpeg']
+const thumbnailSize = Number(env.ATTACHMENTS_THUMBNAIL_SIZE || 250)
+const s3Bucket = String(env.ATTACHMENTS_BUCKET)
+const s3BucketSubdir = String(env.ATTACHMENTS_SUBDIR)
+const s3Region = String(env.ATTACHMENTS_REGION)
+const s3ACL = String(env.ATTACHMENTS_ACL)
+const s3 = new aws.S3()
+const memoryStorage = multer.memoryStorage()
+
+if (env.ATTACHMENTS_BUCKET) {
+  /* eslint-disable-next-line */
+  aws.config.getCredentials(async err => {
+    if (err) {
+      logger.error('Error getting AWS credentials', err.stack || err)
+    } else if (aws.config.credentials) {
+      aws.config.update({
+        secretAccessKey: aws.config.credentials.secretAccessKey,
+        accessKeyId: aws.config.credentials.accessKeyId,
+        region: s3Region
+      })
+    }
+  })
+}
+
+export const multipartFormUpload = multer({ storage: memoryStorage }).single('file')
+
+export function attachmentSummary(attachment: Attachment): AttachmentSummary {
+  const thumbnailUrl = attachment.thumbnail_filename ? attachment.base_url + attachment.thumbnail_filename : ''
+  return {
+    attachment_id: attachment.attachment_id,
+    attachment_url: attachment.base_url + attachment.attachment_filename,
+    thumbnail_url: thumbnailUrl
+  }
+}
+
+export function validateFile(file: SerializedBuffers<Express.Multer.File>) {
+  if (!file || file.buffer.data.length === 0) {
+    throw new ValidationError('No attachment found')
+  } else if (path.extname(file.originalname).replace('.', '') === '') {
+    throw new ValidationError(`Missing file extension in filename ${file.originalname}`)
+  } else if (!supportedMimetypes.includes(file.mimetype)) {
+    throw new UnsupportedTypeError(`Unsupported mime type ${file.mimetype}`)
+  }
```

Comment:
I'm surprised prettier/TS didn't remove all of the `else if`s.
```suggestion
  if (!file || file.buffer.data.length === 0) {
    throw new ValidationError('No attachment found')
  }
  
  if (path.extname(file.originalname).replace('.', '') === '') {
    throw new ValidationError(`Missing file extension in filename ${file.originalname}`)
  } 
  
  if (!supportedMimetypes.includes(file.mimetype)) {
    throw new UnsupportedTypeError(`Unsupported mime type ${file.mimetype}`)
  }
```